dandischema/digests/zarr.py

```diff
@@ -125,9 +127,13 @@ def generate_listing(
                 files=sorted(files) if files is not None else [],
                 directories=sorted(directories) if directories is not None else [],
             )
+        size = sum(file.size for file in checksums.files) + sum(
+            directory.size for directory in checksums.directories
+        )
```

Comment:
yeap, itertools, smth like

```suggestion
        size = sum(
            _.size for _ in itertools.chain(checksums.files, checksums.directories)
        )
```

Reply:
(a) The attributes are lists, so it'd be more efficient to add them than use `chain`; (b) I think this might caught problems if we ever get full type-checking in this library (See #100).

Reply:
> (a) The attributes are lists, so it'd be more efficient to add them than use `chain`;

why?  how creating one more list would be more efficient?

<details>
<summary>some timing shows that chain is efficient</summary> 

```shell
$> ipython3
Python 3.9.10 (main, Jan 16 2022, 17:12:18) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.31.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: files = [f'file{i}' for i in range(1000)]

In [2]: dirs = [f'dir{i}' for i in range(1000)]

In [3]: %timeit sum(len(_) for _ in files + dirs)
147 µs ± 11.2 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [4]: from itertools import chain

In [5]: %timeit sum(len(_) for _ in chain(files, dirs))
138 µs ± 6.17 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [6]: %timeit sum(len(_) for _ in files + dirs)
155 µs ± 15.7 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [7]: %timeit sum(len(_) for _ in chain(files, dirs))
149 µs ± 6.33 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

```
</details> 

but overall -- either way is fine with me.  I just dealing with duplication allergy of having multiple avoidable comprehensions

re type-checking -- again why?  if files and directories entries types are the ones having .size (may be from the same base class even to not rely on duck typing), type checking should then just get smarter since it is IMHO all legit.

Reply:
Regarding the type-checking: I thought the code was using different type for files & directories, but they're actually both `ZarrChecksum`s, so there wouldn't be any problem.

Reply:
I think this is not enough of a bottleneck to benefit from minor optimizations like this, and I think `sum(...) + sum(...)` is more legible an intuitive.